runtime/near-wallet-contract/implementation/wallet-contract/src/internal.rs

```diff
@@ -0,0 +1,286 @@
+use crate::{
+    error::{AccountIdError, CallerError, Error, RelayerError, UserError},
+    eth_emulation, ethabi_utils, near_action,
+    types::{
+        Action, ExecutionContext, TransactionValidationOutcome, ADD_KEY_SELECTOR,
+        ADD_KEY_SIGNATURE, DELETE_KEY_SELECTOR, DELETE_KEY_SIGNATURE, FUNCTION_CALL_SELECTOR,
+        FUNCTION_CALL_SIGNATURE, TRANSFER_SELECTOR, TRANSFER_SIGNATURE,
+    },
+};
+use aurora_engine_transactions::{EthTransactionKind, NormalizedEthTransaction};
+use base64::Engine;
+use ethabi::{ethereum_types::U256, Address};
+use near_sdk::{AccountId, NearToken};
+
+// TODO: Decide on chain id.
+pub const CHAIN_ID: u64 = 0x4ea7;
+const U64_MAX: U256 = U256([u64::MAX, 0, 0, 0]);
+/// Only up to this amount of yoctoNear can be directly mentioned in an action,
+/// the rest should be included in the `value` field of the Ethereum transaction.
+pub const MAX_YOCTO_NEAR: u32 = 1_000_000;
+
+/// The largest accepted `value` field of a transaction.
+/// Computed as `(2**128 - 1) // 1_000_000` since Near balances are
+/// 128-bit numbers, but with 24 decimals instead of 18. So to covert
+/// an Ethereum transaction value into a Near value we need to multiply
+/// by `1_000_000` and then add back any lower digits that were truncated.
+const VALUE_MAX: U256 = U256([10175519178963368024, 18446744073709, 0, 0]);
+
+/// Given an RLP-encoded Ethereum transaction (bytes encoded in base64),
+/// a Near account the transaction is supposed to interact with, the current
+/// account ID, and the current nonce, this function will attempt to transform
+/// the Ethereum transaction into a Near action.
+pub fn parse_rlp_tx_to_action(
+    tx_bytes_b64: &str,
+    target: &AccountId,
+    context: &ExecutionContext,
+    expected_nonce: &mut u64,
```

Comment:
Currently there is no way to delete eth-implicit accounts because the `DeleteAccount` action is disallowed (unsupported action error) by `rlp_execute` and it is also not allowed to add a `FullAccessKey` to the account to get around the wallet contract.

So for now this is not a concern. We'll need to revisit this question if we do allow deleting these accounts in the future.